postgres/datadog_checks/postgres/postgres.py

```diff
@@ -623,7 +623,8 @@ def _new_connection(self, dbname):
             )
             if self._config.query_timeout:
                 connection_string += " options='-c statement_timeout=%s'" % self._config.query_timeout
-            conn = psycopg2.connect(connection_string)
+            conn = psycopg.connect(connection_string, autocommit=True, cursor_factory=psycopg.ClientCursor)
+            conn.read_only = True
```

Comment:
according to this https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#transaction-characteristics-attributes-don-t-affect-autocommit-sessions this will only work if we arent using autocommit